### DISCUSSION TITLE
Pluggable parts of GoReleaser?

### DISCUSSION THREAD

**Discussion by radeksimko** [goreleaser/goreleaser#2584] [Q&A] [answered]
Seeing how GoReleaser is growing in functionality (esp. in the area of announcers and publishers) is great, but also it makes me wonder how much can comfortably be contained and maintained within a single project/repo. Naturally decoupling comes with its own trade-offs, but it still begs a question: Have you considered making some parts of GoReleaser pluggable?

This conversation suggests you did
https://github.com/goreleaser/goreleaser/discussions/1689#discussioncomment-39933

which makes me ask: Have you looked at https://github.com/hashicorp/go-plugin ? 😃  

We use it in most HashiCorp tools (Terraform, Vault, Nomad, Waypoint, Packer) and AFAIK it works on most platforms.

**Comment by caarlos0** [comment, marked as answer]
> which makes me ask: Have you looked at https://github.com/hashicorp/go-plugin ? 😃

I did look at it, but haven't tried it out yet.

I did try using the default go plugin package, but it doesn't work on windows and etc.. but yeah, have things be plugabble would be neat... 

I think we can experiment with it, and if it goes well, cut a v1 on the current state and start working on the pluggable stuff as v2, as I'm sure this would also require config changes 🤔
